extensions/BMO/Extension.pm

```diff
@@ -728,6 +731,23 @@ sub bug_format_comment {
             return qq{<a href="https://hg.mozilla.org/$repo/rev/$id">$text</a>};
         }
     });
+
+    #rewrite mxr with dxr.
+    push (@$regexes, {
+        match => qr/\b(https?:\/\/mxr\.mozilla\.org\S+)\b/,
+        replace => sub {
+            my $args = shift;
+            my $uri = URI->new(html_quote($args->{matches}->[0]));
+            my ($scheme, $auth, $path, $query, $frag);
+            ($scheme, $auth, $path, $query, $frag) = uri_split($uri);
+            my $mark = $uri->query_param_delete("mark");
+            my $rev = $uri->query_param_delete("rev");
+            my $rev = $rev? "?rev=".$rev : '';
+            my $mark = $mark? $mark."," : '';
+            my $hash = $mark || $frag? "#": '';
+            return qq{<a href="$scheme://dxr.mozilla.org$path$rev$hash$mark$frag">$scheme://dxr.mozilla.org$path$rev$hash$mark$frag</a>};
+        }
+    });
```

Comment:
note that code isn't complete:

something else is already in $uri->fragment, we should prepend that if $mark doesn't already contain it. Also most mxr links repeat the line number for single lines, which looks bad: 1234-1234 should replaced by 1234.